pkg/k8shandler/clusterrestart.go

```diff
@@ -0,0 +1,552 @@
+package k8shandler
+
+import (
+	"fmt"
+
+	api "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/pkg/elasticsearch"
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+)
+
+type ClusterRestart struct {
+	client           elasticsearch.Client
+	clusterName      string
+	clusterNamespace string
+	scheduledNodes   []NodeTypeInterface
+}
+
+type Restarter struct {
+	scheduledNodes   []NodeTypeInterface
+	clusterName      string
+	clusterNamespace string
+	clusterStatus    *api.ElasticsearchStatus
+	nodeStatus       *api.ElasticsearchNodeStatus
+
+	precheck func() error
+	prep     func() error
+	main     func() error
+	post     func() error
+	recovery func() error
+
+	precheckCondition func() bool
+	prepCondition     func() bool
+	mainCondition     func() bool
+	postCondition     func() bool
+	recoveryCondition func() bool
+
+	precheckSignaler func()
+	prepSignaler     func()
+	mainSignaler     func()
+	postSignaler     func()
+	recoverySignaler func()
+}
+
+func (elasticsearchRequest *ElasticsearchRequest) PerformFullClusterUpdate(nodes []NodeTypeInterface) error {
```

Comment:
nit. replace receiver fullname `elasticsearchRequest` with `esr`. The current pattern is repeating type names in receiver names.